### ISSUE TITLE
Menus

### ISSUE THREAD

**Issue by sfs-pra** [LBCrion/sfwbar#296]
```
Set Menu1 = "1"
menuclear("menu_name")
menu ("menu_name") {
    item("item", Set Menu1 = "2")
  }
}

layout {
  label {
    value = $Menu1 
    action = menu "menu_name"
  }
}
```
How to make “1” change to “2” when you click on “item”

**Comment by LBCrion** [comment]
With the new functionality you can use expressions in menus,  so:

Set Menu1 = "1"
menuclear("menu_name")
menu ("menu_name") {
    item("item", Eval "Menu1", "2")
  }
}

layout {
  label {
    value = $Menu1
    action = menu "menu_name"
  }
}

**Comment by sfs-pra** [comment]
> With the new functionality you can use expressions in menus, so:

sfwbar-git-r1725.4627ccb - It works
Trying to put it into practice : 
```
Function("SfwbarInit") {
    PipeRead "sfwbar-hwmon.sh"
```
# cat sfwbar-hwmon.sh 
```
#!/bin/sh
echo 'set Hwmon = "'$(ls /sys/class/hwmon/ |tail -n1)'"'
echo '
  file("/sys/class/hwmon/" + $Hwmon + "/temp1_input") {
    CPUTemp = Grab()
  }
'
echo 'menu ("menu_name") {'
for i in `ls /sys/class/hwmon/` ;do
    echo 'item("'$i'", Eval "Hwmon", "'$i'")'
done
echo '}'
```
doesn't work

`item("item save", Exec ("echo 1 > /tmp/sfwbar.ini"))`
It's not working either. 
How do you make the “hwmon” selection?

**Comment by LBCrion** [comment]
Try PipeRead "sh /path/to/sfwbar-hwmon.sh"

Also, can you please post the output of the script?

**Comment by sfs-pra** [comment]
> Try PipeRead "sh /path/to/sfwbar-hwmon.sh"

No improvement

> post the output of the script

```
set Hwmon = "hwmon2"

  file("/sys/class/hwmon/" + $Hwmon + "/temp1_input") {
    CPUTemp = Grab()
  }

menu ("menu_name") {
item("hwmon0", Eval "Hwmon", "hwmon0")
item("hwmon1", Eval "Hwmon", "hwmon1")
item("hwmon2", Eval "Hwmon", "hwmon2")
}
```

**Comment by LBCrion** [comment]
file("/sys/class/hwmon/" + $Hwmon + "/temp1_input")

This won't work yet. Expressions in scanner source declarations aren't
supported yet. It's on the TODO list, but I'll need to rewrite the
variable handling first (a lot of code dealing with the variables is
around 15 years old and really needs an update).

**Comment by sfs-pra** [comment]
Then for now, the only option left is to use menu to configure sfwbar
`item("item save", Exec ("sh -c 'echo 1 > /tmp/sfwbar.ini ; killall -1 sfwbar'"))` - works
```
Function("SfwbarInit") {
    PipeRead "sfwbar-hwmon.sh"
```
```
echo '
  file("/sys/class/hwmon/'$(cat /tmp/sfwbar.ini)'/temp1_input") {
    CPUTemp = Grab()
  }
'
```

**Comment by LBCrion** [comment]
There is a very hacks way to do this without reloading the config:

In your script,  create scanners for all devices, I.e.:

file("/sys/class/hwmon/hwmon0/temp1_input") {
Hwmon0temp = grab()}
file("/sys/class/hwmon/hwmon1/temp1_input") {
Hwmon1temp = grab()}
...

Then declare
Set MyTemp = Hwmon0temp

And in the menu

Item("hwmon1" { config "set mytemp = hwmon1temp"})

And use mytemp as temp value

**Comment by sfs-pra** [comment]
> There is a very hacks way to do this without reloading the config:

Good idea, but it is better to define HwmonXtemp via PipeRead, because on different hardware the amount of Hwmon can be different.

**Comment by sfs-pra** [comment]
There is no %ico in the documentation`item("item1%gtk-ico", Exec "command")`
Is it not possible to add a “tooltip”. Or better yet ItemD("my.desktop")
